### PR TITLE
Changes for 20230915-stable/preview RP API release

### PR DESCRIPTION
**Pull request by Achint-Agrawal**
# Contributing to the Azure SDK

Please see our [CONTRIBUTING.md](https://github.com/Azure/azure-sdk-for-net/blob/main/CONTRIBUTING.md) if you are not familiar with contributing to this repository or have questions.

For specific information about pull request etiquette and best practices, see [this section](https://github.com/Azure/azure-sdk-for-net/blob/main/CONTRIBUTING.md#pull-request-etiquette-and-best-practices).

### CODE REVIEW

**Comment by live1206** [comment]
Please follow below guides to fix the CI:
```
Generated code is not up to date.
    You may need to rebase on the latest main, 
    run 'eng\scripts\Update-Snippets.ps1' if you modified sample snippets or other *.md files (https://github.com/Azure/azure-sdk-for-net/blob/main/CONTRIBUTING.md#updating-sample-snippets), 
    run 'eng\scripts\Export-API.ps1' if you changed public APIs (https://github.com/Azure/azure-sdk-for-net/blob/main/CONTRIBUTING.md#public-api-additions). 
    run 'dotnet build /t:GenerateCode' to update the generated code.
    run 'dotnet build /t:GenerateTests' to update the generated test code.
    
To reproduce this error locally, run 'eng\scripts\CodeChecks.ps1 -ServiceDirectory cosmosdb'.
```
The build failed, please build locally or refer to the [CI](https://github.com/Azure/azure-sdk-for-net/pull/38468/checks?check_run_id=16422077124) log

File: sdk/cosmosdb/Azure.ResourceManager.CosmosDB/src/autorest.md [hunk 1]
```diff
@@ -9,13 +9,18 @@ azure-arm: true
 csharp: true
 library-name: CosmosDB
 namespace: Azure.ResourceManager.CosmosDB
-require: https://github.com/Azure/azure-rest-api-specs/blob/44e83346defd3d4ca99efade8b1ee90c67d9f249/specification/cosmos-db/resource-manager/readme.md
+require: https://github.com/Azure/azure-rest-api-specs/blob/4e7b400ffe0f0d5fabf7b2dcc1e2c65f0be579f5/specification/cosmos-db/resource-manager/readme.md
```
Comment (ArthurMa1978) [inline sdk/cosmosdb/Azure.ResourceManager.CosmosDB/src/autorest.md:12]: The uri must point to the main repo

**Review by ArthurMa1978 — CHANGES_REQUESTED** [review]
Pending on https://github.com/Azure/azure-rest-api-specs/pull/25326

**Review by ajaysnarayan — APPROVED** [review]

File: sdk/cosmosdb/Azure.ResourceManager.CosmosDB/src/autorest.md (hunk 1 above)
Comment (Achint-Agrawal) [reply sdk/cosmosdb/Azure.ResourceManager.CosmosDB/src/autorest.md]: Updated the uri to point to the main repo

File: sdk/cosmosdb/Azure.ResourceManager.CosmosDB/CHANGELOG.md [hunk 2]
```diff
@@ -1,21 +1,29 @@
 # Release History
 
+## Upcoming Release
```
Comment (ArthurMa1978) [inline sdk/cosmosdb/Azure.ResourceManager.CosmosDB/CHANGELOG.md:3]: Dont need these info in the changelog

File: sdk/cosmosdb/Azure.ResourceManager.CosmosDB/CHANGELOG.md [hunk 3]
```diff
@@ -1,21 +1,29 @@
...
+
+- Added support for Priority Based Execution.
+- Added support for cross account container copy.
+- Added new property CustomerManagedKeyStatus to account.
+
 ## 1.4.0-beta.4 (Unreleased)
```
Comment (ArthurMa1978) [inline sdk/cosmosdb/Azure.ResourceManager.CosmosDB/CHANGELOG.md:9]: Please special the expected release date by replacing `Unreleased`, the format is `yyyy-mm-dd`.

File: sdk/cosmosdb/Azure.ResourceManager.CosmosDB/api/Azure.ResourceManager.CosmosDB.netstandard2.0.cs [hunk 4]
```diff
@@ -2052,6 +2029,33 @@ public partial class BackupPolicyMigrationState
         public static bool operator !=(Azure.ResourceManager.CosmosDB.Models.BackupPolicyType left, Azure.ResourceManager.CosmosDB.Models.BackupPolicyType right) { throw null; }
         public override string ToString() { throw null; }
     }
+    public partial class BackupSchedule
```
Comment (ArthurMa1978) [inline sdk/cosmosdb/Azure.ResourceManager.CosmosDB/api/Azure.ResourceManager.CosmosDB.netstandard2.0.cs:2032]: The name is too generic, consider to rename it with a more descriptive and meaningful name.

File: sdk/cosmosdb/Azure.ResourceManager.CosmosDB/api/Azure.ResourceManager.CosmosDB.netstandard2.0.cs [hunk 5]
```diff
@@ -2052,6 +2029,33 @@ public partial class BackupPolicyMigrationState
...
+    {
+        public BackupSchedule() { }
+        public string CronExpression { get { throw null; } set { } }
+        public int? RetentionInHours { get { throw null; } set { } }
+        public string ScheduleName { get { throw null; } set { } }
+    }
+    [System.Runtime.InteropServices.StructLayoutAttribute(System.Runtime.InteropServices.LayoutKind.Sequential)]
+    public readonly partial struct BackupState : System.IEquatable<Azure.ResourceManager.CosmosDB.Models.BackupState>
```
Comment (ArthurMa1978) [inline sdk/cosmosdb/Azure.ResourceManager.CosmosDB/api/Azure.ResourceManager.CosmosDB.netstandard2.0.cs:2040]: The name is too generic, consider to rename it with a more descriptive and meaningful name.

File: sdk/cosmosdb/Azure.ResourceManager.CosmosDB/api/Azure.ResourceManager.CosmosDB.netstandard2.0.cs [hunk 6]
```diff
@@ -2359,6 +2376,24 @@ public partial class CheckCosmosDBNameAvailabilityResponse
         public Azure.ResourceManager.CosmosDB.Models.CosmosDBNameUnavailableReason? Reason { get { throw null; } }
     }
     [System.Runtime.InteropServices.StructLayoutAttribute(System.Runtime.InteropServices.LayoutKind.Sequential)]
+    public readonly partial struct ClusterType : System.IEquatable<Azure.ResourceManager.CosmosDB.Models.ClusterType>
```
Comment (ArthurMa1978) [inline sdk/cosmosdb/Azure.ResourceManager.CosmosDB/api/Azure.ResourceManager.CosmosDB.netstandard2.0.cs:2379]: The name is too generic, consider to rename it with a more descriptive and meaningful name.

File: sdk/monitor/Azure.Monitor.OpenTelemetry.Exporter/api/Azure.Monitor.OpenTelemetry.Exporter.net6.0.cs
Comment (ArcturusZhang) [inline sdk/monitor/Azure.Monitor.OpenTelemetry.Exporter/api/Azure.Monitor.OpenTelemetry.Exporter.net6.0.cs:1]: despite this file should be included in this RP, we should remove it in this PR because this is not the purpose of this PR.

You could run the `Export-API` script in this way to avoid generating api files in other RPs:
```
./eng/scripts/Export-API.ps1 cosmosdb
```

File: sdk/servicebus/Microsoft.Azure.WebJobs.Extensions.ServiceBus/api/Microsoft.Azure.WebJobs.Extensions.ServiceBus.net6.0.cs
Comment (ArcturusZhang) [inline sdk/servicebus/Microsoft.Azure.WebJobs.Extensions.ServiceBus/api/Microsoft.Azure.WebJobs.Extensions.ServiceBus.net6.0.cs:1]: same here, please revert this change in this PR

File: sdk/cosmosdb/Azure.ResourceManager.CosmosDB/src/Generated/Models/ThroughputSettingsResourceInfo.cs [hunk 7]
```diff
@@ -36,5 +40,9 @@ internal ThroughputSettingsResourceInfo(int? throughput, AutoscaleSettingsResour
         public string MinimumThroughput { get; }
         /// <summary> The throughput replace is pending. </summary>
         public string OfferReplacePending { get; }
+        /// <summary> The offer throughput value to instantly scale up without triggering splits. </summary>
+        public string InstantMaximumThroughput { get; }
```
Comment (archerzz) [inline sdk/cosmosdb/Azure.ResourceManager.CosmosDB/src/Generated/Models/ThroughputSettingsResourceInfo.cs:44]: `Throughput` is `int?`, why other `XXXThroughput` are `string`?

**Review by archerzz — CHANGES_REQUESTED** [review]
Some classes are removed (like `CassandraClusterBuackResource`), which will cause breaking change. Is that expected? Note that normally we should not break customers' codes.

File: sdk/cosmosdb/Azure.ResourceManager.CosmosDB/CHANGELOG.md (hunk 2 above)
Comment (Achint-Agrawal) [reply sdk/cosmosdb/Azure.ResourceManager.CosmosDB/CHANGELOG.md]: Removed

File: sdk/cosmosdb/Azure.ResourceManager.CosmosDB/CHANGELOG.md (hunk 3 above)
Comment (Achint-Agrawal) [reply sdk/cosmosdb/Azure.ResourceManager.CosmosDB/CHANGELOG.md]: Added 2023-10-30 as a placeholder. Might change this after discussing with RP team

File: sdk/cosmosdb/Azure.ResourceManager.CosmosDB/api/Azure.ResourceManager.CosmosDB.netstandard2.0.cs (hunk 5 above)
Comment (Achint-Agrawal) [reply sdk/cosmosdb/Azure.ResourceManager.CosmosDB/api/Azure.ResourceManager.CosmosDB.netstandard2.0.cs]: Changed to CassandraClusterBackupState

File: sdk/cosmosdb/Azure.ResourceManager.CosmosDB/api/Azure.ResourceManager.CosmosDB.netstandard2.0.cs (hunk 4 above)
Comment (Achint-Agrawal) [reply sdk/cosmosdb/Azure.ResourceManager.CosmosDB/api/Azure.ResourceManager.CosmosDB.netstandard2.0.cs]: Changed to CassandraClusterBackupSchedule

File: sdk/monitor/Azure.Monitor.OpenTelemetry.Exporter/api/Azure.Monitor.OpenTelemetry.Exporter.net6.0.cs
Comment (Achint-Agrawal) [reply sdk/monitor/Azure.Monitor.OpenTelemetry.Exporter/api/Azure.Monitor.OpenTelemetry.Exporter.net6.0.cs:1]: Removed changes outside cosmosdb

File: sdk/servicebus/Microsoft.Azure.WebJobs.Extensions.ServiceBus/api/Microsoft.Azure.WebJobs.Extensions.ServiceBus.net6.0.cs
Comment (Achint-Agrawal) [reply sdk/servicebus/Microsoft.Azure.WebJobs.Extensions.ServiceBus/api/Microsoft.Azure.WebJobs.Extensions.ServiceBus.net6.0.cs:1]: Removed changes outside cosmosdb

File: sdk/cosmosdb/Azure.ResourceManager.CosmosDB/api/Azure.ResourceManager.CosmosDB.netstandard2.0.cs (hunk 6 above)
Comment (Achint-Agrawal) [reply sdk/cosmosdb/Azure.ResourceManager.CosmosDB/api/Azure.ResourceManager.CosmosDB.netstandard2.0.cs]: Changed to CassandraClusterType

**Review by archerzz — COMMENTED** [review]
OK. The deleted classes are from beta verions.

**Review by archerzz — APPROVED** [review]

**Comment by pjohari-ms** [comment]
@archerzz Is there a documentation on how we can release the beta SDK separate from stable SDK?

**Comment by pjohari-ms** [comment]
Since the source code is the same, we're forced to release one single SDK for the preview API version. But from your comment it seems like the beta/preview version process is different.

**Comment by archerzz** [comment]
> @archerzz Is there a documentation on how we can release the beta SDK separate from stable SDK?

@pjohari-ms Release a beta version is the same as releasing a stable version, except for the version in metadata. See this PR as an example: https://github.com/Azure/azure-sdk-for-net/pull/39215

**Review by ArthurMa1978 — APPROVED** [review]
